docs/manual/scala/guide/services/DependencyInjection.md

```diff
@@ -0,0 +1,55 @@
+
+@[lagom-loader](code/ServiceImplementation.scala)
+
+The loader has two methods that must be implemented, `load` and `loadDevMode`. You can see that we've mixed in different service locators for each method, we've mixed in [`LagomDevModeComponents`](api/com/lightbend/lagom/scaladsl/devmode/LagomDevModeComponents.html) that provides the dev mode service locator and registers the services with it in dev mode, and in prod mode, for now, we've simply provided [`NoServiceLocator`](api/com/lightbend/lagom/scaladsl/api/ServiceLocator$$NoServiceLocator$.html) as the service locator - this is a service locator that will return nothing for every lookup. We'll see in the [[deploying to production|ProductionOverview]] documentation how to select the right service locator for production.
+
+A third method, `describeServices`, is optional, but may be used by tooling, for example by [[ConductR]], to discover what service APIs are offered by this service. The meta data read from here may in turn be used to configure service gateways and other components.
+
+Finally, we need to tell Play about our application loader. We can do that by adding the following configuration to `application.conf`:
+
+    play.application.loader = com.example.HelloLoader
+
+
+## Defining your own components
+
+To create a complex Application (one that uses Persistence, Clustering, the Broker API, etc...) you will need to mix in many Components. It is a good choice to create small custom traits that mix in some of those Components and build the Application by mixing in your small custom traits. That will let you test parts of the complete Application in isolation.
+
+Imagine your Service consumes messages from a Broker topic where `Orders` are notified. Then your service stores that info into a database and does some processing with a final step of invoking a third party endpoint. If you only wanted to test the consuming of messages and proper storage you could create an `OrderConsumingComponent` trait and mix in `LagomServiceClientComponents` and `CassandraPersistenceComponents` so that you could consume the messages and store them. On your test you could extend your `OrderConsumingComponent` with `TestTopicComponents` that provides a mocked up broker so you didn't need to start a broker to run the tests. Finally, on your Application you would mix in the tested `OrderConsumingComponent` and `LagomKafkaClientComponents`.
+
+Lagom provides several Components. For this reference guide we grouped the components by feature:
+
+ * [[Service Components|ScalaComponents#Service-Components]]
+ * [[Persistence and Cluster Components|ScalaComponents#Persistence-and-Cluster-Components]]
+ * [[Broker API Components|ScalaComponents#Broker-API-Components]]
+ * [[Service Locator Components|ScalaComponents#Service-Locator-Components]]
+ * [[Third party Components|ScalaComponents#Third-party-Components]]
```

Comment:
As @rstento suggested, the DI page ends on a list of quick links to each group of components. Since this is already provided I made it so the _next:_ of DI would be `Service Metadata` (instead of `Scala Components`.

Reading the guide top to bottom (or in the right-hand side index) the user will see:

- Service descriptors
- Implementing services
- Dependency Injection
- Scala Components
- Service Metadata
- Consuming services
- Testing Services
- Message serializers
- Header Filters
- Error handling

But in the `index.toc` I set it so `Dependency Injection`'s next was not  `Scala Components` but `Service Metadata`.